beesdoo_pos/models/product.py

```diff
@@ -0,0 +1,16 @@
+from odoo import api, fields, models, _
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.onchange('sale_ok')
+    def _onchange_sale_ok(self):
+        """ Maintains the value of `available_in_pos` when unchecking `sale_ok`,
+            by storing its value before calling parent's method,
+            then assigning the stored value.
+        """
+        is_available = self.available_in_pos
+        super(ProductTemplate, self)._onchange_sale_ok()
+        if not self.sale_ok:
+            self.available_in_pos = is_available
```

Comment:
Have you tested that a product with `sale_ok` False and `available_in_pos` True is realy not available in pos. My fear is that the POS may rely only in the `available_in_pos` value to check if the product can appear in the POS regardless of the value of `sale_ok`.

If you tested this case, this "naive" approach may be good. If not the previous value of `available_in_pos` should be stored in a specific field in database to be restored properly.

Reply:
I've tested the case : 

1. a product has `sale_ok = False` and `available_in_pos = True`
2. opened a POS session, said product doesn't appear 

=> a product appear in the POS seems to be depending on the value of `sale_ok`